pkg/registry/core/rest/storage_core.go

```diff
@@ -296,10 +346,23 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(apiResourceConfigSource
 		podStorage.Pod,
 		c.ProxyTransport)
 	if err != nil {
-		return LegacyRESTStorage{}, genericapiserver.APIGroupInfo{}, err
+		return legacyRESTStorage{}, genericapiserver.APIGroupInfo{}, err
+	}
+
+	storage := apiGroupInfo.VersionedResourcesStorageMap["v1"]
+	if storage == nil {
+		storage = map[string]rest.Storage{}
+	}
+
+	// potentially override the generic serviceaccount storage with one that supports pods
```

Comment:
this is the only non-code move change, overriding the generic serviceaccount storage with wired pod getter. Not super nice, but cannot come up with a better pattern.